packages/vite-plugin/src/partials/editor.ts

```diff
@@ -20,15 +20,24 @@ export function editorPlugin({editor, projects}: EditorPluginConfig): Plugin {
 
   const lookup = new Map<string, ProjectData>();
   for (const project of projects) {
-    lookup.set(project.name, project);
+    lookup.set(project.filePath, project);
   }
 
+  let query: string;
+
   return {
     name: 'motion-canvas:editor',
 
-    async load(id) {
-      const [, query] = id.split('?');
+    async resolveId(source) {
+      if (source.startsWith(resolvedEditorId)) {
+        const [id, projectQuery] = source.split('?');
+        query = projectQuery;
+        return id;
+      }
+      return null;
+    },
```

Comment:
This is a bit too hacky. There's no guarantee the corresponding `load` will be called right after `resolveId`. The same server may be used to serve two projects at the same time, for example.
Also, this global `query` variable will break the project selections screen:
1. Visit localhost:9000
2. Choose one of the projects
3. Go back to localhost:9000
4. Instead of the selection screen, the previously chosen project is immediately loaded

The reason you were getting an error was because the name was not escaped when generating the `virtual:editor?project=[name]` url. All you need to do is add `encodeURIComponent` in the server a few lines below:
```ts
if (name && lookup.has(name)) {
  res.setHeader('Content-Type', 'text/html');
  res.end(
    createHtml(
      // right here:
      `/@id/__x00__virtual:editor?project=${encodeURIComponent(
        name,
      )}`,
    ),
  );
  return;
}
```
and it will work just fine without `resolveId` and the global `query` variable.

Reply:
Ah. Ok got it.